devel/refresh_cache.sh

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/bash -ex
+# Copyright (c) 2017 Red Hat, Inc.
+#
+# This file is part of Bodhi.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+# This script is used to run the tests in a variety of parallel
+# containers to test Bodhi across supported Fedora releases (See the RELEASES variable below). You
+# can set the RELEASES environment variable to override the RELEASES for a given test run if you
+# like. You can pass a -x flag to it to get it to exit early if a build or test run fails.
+# It is intended to be run with sudo, since it needs to use docker.
+
+RELEASES=${RELEASES:="f25 f26 f27 rawhide pip"}
+
+PARALLEL="parallel -v --tag"
+
+# Assemble the Dockerfile snippets into Dockerfiles for each release.
+pushd devel/ci/
+for r in $RELEASES; do
+    cat Dockerfile-header > Dockerfile-$r
+    if [ $r != "pip" ]; then
+        # Add the common rpm packages for the non-pip releases.
+        cat rpm-packages >> Dockerfile-$r
+    else
+        # Let's use F26 for the pip tests.
+        sed -i "s/FEDORA_RELEASE/26/" Dockerfile-$r
+    fi
+    cat $r-packages Dockerfile-footer >> Dockerfile-$r
+    echo "COPY . /bodhi" >> Dockerfile-$r
+done
+popd
```

Comment:
This block seems the same as the one in ```run_tests.sh``` - should we break it out into something shared? I was actually thinking about using a ```Makefile``` to do this as you had suggested to me.